src/forms-opportunities/form-details-handler/detect-form-details.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { ok } from '@adobe/spacecat-shared-http-utils';
+
+export default async function handler(message, context) {
+  const {
+    log, dataAccess, sqs, env,
+  } = context;
+  const { Opportunity } = dataAccess;
+  const { siteId, data } = message;
+  log.info(`Message received in form details handler: ${JSON.stringify(message, null, 2)}`);
+  const {
+    url, form_source: formsource, form_details: formDetails, opportunity_type: opportunityType,
+  } = data;
+
+  const existingOpportunities = await Opportunity.allBySiteId(siteId);
+  const opportunity = existingOpportunities
+    .filter((oppty) => oppty.getType() === opportunityType)
+    .find((oppty) => oppty.getData()?.form === url && (!formsource
+          || oppty.getData()?.formsource === formsource));
+
+  if (opportunity) {
+    log.info(`Opportunity found: ${JSON.stringify(opportunity)}`);
+    opportunity.setUpdatedBy('system');
+    opportunity.setData({
+      ...opportunity.getData(),
+      ...formDetails,
+    });
```

Comment:
I think we should skip the urls where lead gen form is not there.. but first test the accuracy of lead gen form identification.

Reply:
that can be added in the backlog and should be handled separately